### DISCUSSION TITLE
Have to refresh for browser to continue to render components

### DISCUSSION THREAD

**Discussion by KadeerAmin** [mswjs/msw#1177] [Q&A] [answered]
Hi,

I'm trying to use msw to mock rest requests in the browser to create a demo environment and I have followed the setup steps.

Everything is set up per the guides and I can see `Status Code: 200 OK (from service worker)` in the network tab for requests and warnings for the unhandled requests for things like static assets.

The page seemingly needs to be refreshed each time the URL changes for the react components to continue rendering.

Video for context. Note the URL changes each time I refresh.

https://user-images.githubusercontent.com/21316846/159587958-255798d6-f771-4711-994e-f19b766716e4.mov

I was hoping for help troubleshooting this.

**Comment by kettanaito** [comment, marked as answer]
Hey, @KadeerAmin. 

Do you have a reproduction repository where I can look into this? It doesn't seem to be an issue with MSW. Could you please take the following steps to be sure:

1. Update to the latest version of the library `npm i msw@latest`. Also, don't forget to update the worker script if you're not using `msw.workerDirectory` property in your package.json.
2. If the issue persists, disable MSW and see if the issue is still reproducible. If it is, it's likely the root cause lies somewhere in your app's code. If it's solved, then I'd kindly ask for a reproduction repository.

This kind of issue is possible conceptually if the worker intercepts static assets. However, we have an explicit logic in the worker script to ignore static assets and I believe it's functioning correctly.

**Reply by KadeerAmin** [reply]
So sorry to bother you with this. My issues with msw worker were a red herring.

While trying to replicate the repository I realized that I had accidentally updated my `history` package to v5 which was incompatible with my version of `react-router-dom` causing the issue.

I nonetheless appreciate the timely response!
